gateway/dataimpl/server_v1/queryadminserver.go

```diff
@@ -89,6 +91,67 @@ func (s *QueryIndexAdminServer) executeQuery(
 	return rows, nil
 }
 
+func (s *QueryIndexAdminServer) validateNames(index, bucket, scope, col *string) error {
+	var missingType string
+	switch {
+	case bucket != nil && *bucket == "":
+		missingType = "Bucket"
+	case scope != nil && *scope == "":
+		missingType = "Scope"
+	case col != nil && *col == "":
+		missingType = "Collection"
+	}
+
+	if index != nil {
+		if strings.ContainsAny(*index, `!,.$%^&*()+={}[]':;|\<>?@`) {
+			return s.errorHandler.NewQueryIndexInvalidArgumentStatus(
+				nil,
+				*index,
+				"Index name cannot contain special characters").Err()
+		}
+
+		if len(*index) >= 220 {
+			return s.errorHandler.NewQueryIndexInvalidArgumentStatus(
+				nil,
+				*index, "Index name cannot be longer than 219 characters.").Err()
+		}
+
+		if *index == "" {
+			return s.errorHandler.NewQueryIndexInvalidArgumentStatus(
+				nil,
+				*index, "Index name cannot an empty string.").Err()
+		}
+
+		if strings.Contains(*index, " ") {
+			return s.errorHandler.NewQueryIndexInvalidArgumentStatus(
+				nil,
+				*index, "Index name cannot contain spaces.").Err()
+		}
+	}
+
+	if col != nil && strings.Contains(*col, " ") {
+		return s.errorHandler.NewQueryIndexInvalidArgumentStatus(
```

Comment:
At what point do we draw the line at checking invalid args? If we're doing this for collection then should we be doing it for bucket and scope too?

Reply:
The server didn't return errors when the bucket/scope contains spaces so I didn't add validation for it, when collection name contains a space we get:
```
{"level":"error","ts":"2023-11-23T11:39:44.071Z","logger":"gateway.data-impl.errors","caller":"server_v1/errorhandler.go:636","msg":"handling generic error","error":"query server error: query error: parsing failure (code: 3000, msg: Invalid identifier '_default '
```
Whereas with a bucket and scope that contains a space we respectively get:
```
{"level":"error","ts":"2023-11-23T11:42:46.830Z","logger":"gateway.data-impl.errors","caller":"server_v1/errorhandler.go:636","msg":"handling generic error","error":"rpc error: code = NotFound desc = Bucket 'default ' was not found.","stacktrace":"github.com/couchbase/stellar-gateway/gateway/dataimpl/server_v1.ErrorHandler.NewGenericStatus

{"level":"error","ts":"2023-11-23T11:41:00.568Z","logger":"gateway.data-impl.errors","caller":"server_v1/errorhandler.go:636","msg":"handling generic error","error":"query server error: query error: index failure (code: 12021, msg: Scope not found in CB datastore default:default._default )
```

But given that we can't create buckets/scopes/collections with spaces in the name perhaps it would be sensible to validate them?

Reply:
I made the decision to add validation for bucket and scope names containing spaces